### PR TITLE
[4/12] Improves coverage for src/dom.js by testing function waitForDocumentBody 

### PR DESCRIPTION
**Pull request by Cijin**
* In `setup.js`, document.body property mocked to make `document.body = null`. This wasn't possible without mocking as `null` is not a html element

* Line 70 left untested, as throw is unreachable as discussed in issue #33.

### CODE REVIEW

File: test/tests/dom/waitForDocumentBody.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    beforeEach(memoize.clear);
+
+    afterEach(() => {
+        document.body = oldBody;
+    });
+    it('should resolve when body is present', async () => {
+        try {
+            document.readyState = 'complete';
+            document.body = testBody;
+            await waitForDocumentBody();
```
Comment (bluepnume) [inline test/tests/dom/waitForDocumentBody.js:19]: Can you also check the return value is `=== testBody`?

File: test/tests/dom/waitForDocumentBody.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        } catch (err) {
```
Comment (bluepnume) [inline test/tests/dom/waitForDocumentBody.js:20]: I think we can remove this try/catch -- if there's an error thrown, it's probably best for the test to fail with the original error.

File: test/tests/dom/waitForDocumentBody.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            throw new Error('Expected waitForDocumentReady to resolve');
+        }
+    });
+
+    it('should eventully resolve when document is ready', async () => {
```
Comment (bluepnume) [inline test/tests/dom/waitForDocumentBody.js:25]: (same comments for this test)

**Comment by bluepnume** [comment]
Looks great, thanks! Just a few minor comments then will merge this one.

File: test/tests/dom/waitForDocumentBody.js (hunk 1 above)
Comment (Cijin) [reply test/tests/dom/waitForDocumentBody.js:19]: yes ofcourse

**Review by Cijin — COMMENTED** [review]
Hey, I've updated the pr :)

**Comment by bluepnume** [comment]
Awesome! Thank you!

**Pull request merged**